pm_mac/pmmacosxcm.c

```diff
@@ -963,6 +972,47 @@ static unsigned int midi_check_host_error(PmInternal *midi)
     return FALSE;
 }
 
+UInt64 current_host_time(void)
+{
+#if TARGET_OS_OSX
+    return AudioGetCurrentHostTime();
+#else
+    return mach_absolute_time();
+#endif
+}
```

Comment:
The mach methods are available on macOS too, so we could probably drop the macOS-specific `AudioGetCurrentHostTime`, `AudioGetHostClockFrequency` etc. I've kept them around under a `TARGET_OS_OSX` conditional for now to avoid accidentally introducing a regression if there's a bug in the (albeit quite short) mach-based implementation for iOS. Since it introduces an additional, likely not very heavily used code path, I would also be open to removing the macOS branch again. Thoughts welcome.

Reply:
Apple documentation says it is preferred to use clock_gettime_nsec_np(CLOCK_UPTIME_RAW), but doesn't say why. Is there some reason to choose mach_absolute_time()?

I'm not sure why AudioGetCurrentHostTime was used, but I've benchmarked it because I was worried about the cost given how often it gets called. I forgot the numbers, but remember it's quite fast. I would want to compare execution speed before switching the implementation for OSX, and the TARGET_OS_OSX seems best for now.

Reply:
> Is there some reason to choose mach_absolute_time()?

It seems to be the officially recommended drop-in replacement: https://developer.apple.com/library/archive/qa/qa1643/_index.html

Reply:
`clock_gettime_nsec_np(CLOCK_UPTIME_RAW)` seems to be a wrapper around `mach_absolute_time`:

https://github.com/apple-oss-distributions/Libc/blob/c5a3293354e22262702a3add5b2dfc9bb0b93b85/gen/clock_gettime.c#L117-L119

So it's probably a question of style which one to choose? `mach_absolute_time` seems pretty widely used, so I'm sure they won't deprecate that anytime soon.

Reply:
Interesting, because [documentation for mach_absolute_time](https://developer.apple.com/documentation/kernel/1462446-mach_absolute_time) says use clock_gettime_nsec_np(CLOCK_UPTIME_RAW) (!). But OK, if Apple can't decide, I'll go with your choice.

Reply:
Thanks for the link, I must have missed that note. Interesting that they mention fingerprinting. But regardless, `mach_absolute_time` is the lower level API and almost all projects I've seen that use Core Audio/Core MIDI on iOS went with `mach_absolute_time` too.